### PR TITLE
Add minimal CLI

### PR DESCRIPTION
**Pull request by stephenfin**
Add a minimal CLI interface that can be used as an alternative to `build`.

This clearly isn't complete: I'd like to see whether it's worth doing before I go any further.

### CODE REVIEW

File: src/pyproject_api/build.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import argparse
```
Comment (gaborbernat) [inline src/pyproject_api/build.py:1]: Should be `__main__` instead 😊

File: src/pyproject_api/build.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import os
```
Comment (gaborbernat) [inline src/pyproject_api/build.py:2]: Adf `from __future__ import annotations`.

File: src/pyproject_api/build.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    parser.add_argument(
+        "--outdir",
+        "-o",
+        type=pathlib.Path,
+        help=f"output directory (defaults to {{srcdir}}{os.sep}dist)",
+    )
+    return parser
+
+
+def build(argv: list[str]) -> None:
```
Comment (gaborbernat) [inline src/pyproject_api/build.py:45]: name it main instead of build

**Review by gaborbernat — CHANGES_REQUESTED** [review]
You'll need to add tests too, and I think we should add build_editable too via `-e`, `--editable` too.

File: src/pyproject_api/__main__.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        description=textwrap.indent(
+            textwrap.dedent(
+                """A minimal PEP-517 frontend.
+
+                This is mainly useful for debugging PEP-517 backends. This frontend will not
+                actually install required build dependencies.
+                """
+            ).strip(),
+            "    ",
+        ),
```
Comment (gaborbernat) [inline src/pyproject_api/__main__.py:24]: I don't want to replicate their desc. Let's replace this with `pyproject.toml file based build frontend`.

File: tests/test_main.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    subprocess_frontend.build_sdist.return_value = SdistResult(sdist=out / "foo.whl", out="", err="")
+    subprocess_frontend.build_wheel.return_value = WheelResult(wheel=out / "foo.whl", out="", err="")
+    subprocess_frontend.build_editable.return_value = EditableResult(wheel=out / "foo.whl", out="", err="")
+    mocker.patch("pyproject_api.__main__.SubprocessFrontend", return_value=subprocess_frontend)
+
+    pyproject_api.__main__.main(cli_args)
+
+    # FIXME: Something weird is going on here and I can't figure out what. I suspect it's something to do with us
+    # mocking the whole class and this being a classmethod
+    # subprocess_frontend.create_args_from_folder.assert_called_once_with(srcdir)
```
Comment (gaborbernat) [inline tests/test_main.py:62]: We should fix this before merging

**Review by gaborbernat — CHANGES_REQUESTED** [review]

File: tests/test_main.py (hunk 5 above)
Comment (stephenfin) [reply tests/test_main.py:62]: Figured it out. Class != object. Duh :see_no_evil:

**Comment by stephenfin** [comment]
I'm still not sure about the `-e / --editable`. Given we're not actually installing this thing, what value is there is generating them? From [PEP-660](https://peps.python.org/pep-0660/#build-editable):

> An “editable” wheel uses the wheel format not for distribution but as ephemeral communication between the build system and the front end. This avoids having the build backend install anything directly. This wheel must not be exposed to end users, nor cached, nor distributed.

Should we really be exposing this?

File: src/pyproject_api/__main__.py [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+
+from ._frontend import EditableResult, SdistResult, WheelResult
+from ._via_fresh_subprocess import SubprocessFrontend
+
+
+def main_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(
+        description=textwrap.indent(
+            textwrap.dedent(
+                """A pyproject.toml-based build frontend.
```
Comment (gaborbernat) [inline src/pyproject_api/__main__.py:17]: Can we keep this one line please?

**Review by gaborbernat — CHANGES_REQUESTED** [review]

File: src/pyproject_api/__main__.py (hunk 6 above)
Comment (stephenfin) [reply src/pyproject_api/__main__.py:17]: We can, but is the rest of the context not helpful? If anything, I thought we'd want to expand it further.

Comment (gaborbernat) [reply src/pyproject_api/__main__.py:17]: We can keep the words, just let the argparser break it up, no? Rather than us formatting it.

Comment (stephenfin) [reply src/pyproject_api/__main__.py:17]: Ohhh, gotcha :+1: Coming right up...

**Review by gaborbernat — APPROVED** [review]

**Comment by gaborbernat** [comment]
By the way what's your use case for this functionality?

**Pull request merged**